File: CPP/Petshop.cpp
#include <iostream>
#include <string>

using namespace std;

class Petshop{
	private:
		int ID;
		string Nama_Produk;
		string Kategori_Produk;
		int Harga;
		
	public:
		// Konstruktor (untuk menambah objek)
		Petshop(){
		}
		
		Petshop(int id, string nama_produk, string kategori_produk, int harga){
			ID = id;
			Nama_Produk = nama_produk;
			Kategori_Produk = kategori_produk;
			Harga = harga;
		}
		
		// Setter (untuk mengubah data objek & menambahkan data baru jika kosong)
		void setId(int id){ID = id;}
		void setNamaProduk(string nama_produk){Nama_Produk = nama_produk;}
		void setKategoriProduk(string kategori_produk){Kategori_Produk = kategori_produk;}
		void setHarga(int harga){Harga = harga;}
		
		// Getter (untuk mencari data yang diinginkan berdasarkan produk)
		int getId(){return ID;}
		string getNamaProduk(){return Nama_Produk;}
		string getKategoriProduk(){return Kategori_Produk;}
		int getHarga(){return Harga;}
		
		// Destruktor
		~Petshop(){
		}
};

File: Java/Petshop.java
class Petshop{
    private int ID = 0;
    private String Nama_Produk = "";
    private String Kategori_Produk = "";
    private int Harga = 0;

    // Konstruktor
    public Petshop(int id, String nama_produk, String kategori_produk, int harga) {
        this.ID = id;
        this.Nama_Produk = nama_produk;
        this.Kategori_Produk = kategori_produk;
        this.Harga = harga;
    }

    // Setter
    void setId(int id){this.ID = id;}
    void setNamaProduk(String namaProduk){this.Nama_Produk = namaProduk;}
    void setKategoriProduk(String kategoriProduk){this.Kategori_Produk = kategoriProduk;}
    void setHarga(int harga){this.Harga = harga;}

    // Getter
    int getId(){return this.ID;}
    String getNamaProduk(){return this.Nama_Produk;}
    String getKategoriProduk(){return this.Kategori_Produk;}
    int getHarga(){return this.Harga;}
}
